src/psyclone/psyGen.py

```diff
@@ -381,10 +381,12 @@ class Invoke(object):
     ''' Manage an individual invoke call '''
 
     def __str__(self):
-        return self._name+"("+str(self.unique_args)+")"
+        return self._name+"("+", ".join(map(lambda x:str(x), self._alg_unique_args))+")"
 
     def __init__(self, alg_invocation, idx, schedule_class,
                  reserved_names=None):
+        self._name = "invoke"
+        self._alg_unique_args = []
```

Comment:
New lines are covered and make sense. However line 384 is not pep8-compliant.

Reply:
Having said that, pylint thinks the `lambda` is unnecessary. Perhaps use a list comprehension:

    ", ".join([str(arg) for arg in self._alg_unique_args])